controller/service.go

```diff
@@ -281,7 +313,10 @@ func getDesiredLbIPs(svc *v1.Service) ([]net.IP, ipfamily.Family, error) {
 		return nil, "", fmt.Errorf("invalid spec.loadBalancerIP %q", svc.Spec.LoadBalancerIP)
 	}
 	desiredLbIPs = append(desiredLbIPs, desiredLbIP)
-	desiredLbIPFamily := ipfamily.ForAddress(desiredLbIP)
+	desiredLbIPFamily, err := ipfamily.ForAddressesIPs(desiredLbIPs, familyPolicy)
+	if err != nil {
+		return nil, "", err
+	}
 
```

Comment:
does it make sense here to have a validation that if the familyPolicy is requireDualStack and user provided a single stack annotation we fail?

Reply:
I'm not sure why would we want that? `ForAddressesIPs` should handle that situation.

Reply:
then ignore me (for now 😄), didn't go over that part of the code yet (sorry I should have mentioned what I did go over)